### PR TITLE
Video block: Add poster image

### LINKED ISSUE #4837: Support of poster image for video block is missing
At the moment, there's no option to set the ~~cover~~ poster image for a video.

Tested with Gutenberg 2.1

### PR DESCRIPTION
**Pull request by Soean**
## Description
This PR adds the ability to add a poster image to the video block.

Fixes #4837

## Screenshots
No poster selected
bildschirmfoto 2018-08-24 um 22 32 41

Poster selected
bildschirmfoto 2018-08-24 um 22 32 29

Workflow
video-poster

### CODE REVIEW

**Comment by jasmussen** [comment]
This is pretty cool. 

Seeing as this is not part of this PR, separately we should consider adding contextual help to the switches. Their effect is boolean and instant so it's okay to use switches, but we should have some help text below.

File: packages/block-library/src/video/edit.js [hunk 1]
```diff
@@ -74,8 +79,28 @@ class VideoEdit extends Component {
 		this.setState( { editing: false } );
 	}
 
+	onSelectPoster( image ) {
+		const { setAttributes, clientId } = this.props;
+		setAttributes( { poster: image.url } );
+		document.getElementById( 'block-' + clientId ).getElementsByTagName( 'video' )[ 0 ].load();
```
Comment (gziolo) [inline packages/block-library/src/video/edit.js]: I think we should use `ref` in this case. See the related docs: https://reactjs.org/docs/refs-and-the-dom.html

Usage example in the existing code:
https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/block/edit-panel/index.js#L15
https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/block/edit-panel/index.js#L66
https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/block/edit-panel/index.js#L35

Comment (gziolo) [reply packages/block-library/src/video/edit.js]: By the way, this logic should probably be moved to `componenDidUpdate` lifecycle method as it should be triggered before `this.props.attributes.poster` isn't updated.

**Review by SergioEstevao — COMMENTED** [review]
I will leave the code review for the experts, but in terms of the output/input HTML it's what we are expecting in Aztec. So looking great!

**Comment by gziolo** [comment]
@SergioEstevao, awesome to hear that. By the way, are there any other missing bits to make Gutenberg fully compatible in Aztec? This one was the only one I was aware of.

**Comment by Soean** [comment]
@gziolo Thanks for your review. This is a better much better solution, I updated this PR.

**Review by gziolo — APPROVED** [review]
This works great, thanks for adding the fix for the `ref`.

Would it be possible to improve the way poster is displayed? This is how it looks for my test video + image:

screen shot 2018-08-29 at 09 58 43

**Comment by gziolo** [comment]
I added fix for the `poster` issue I shared. Kudos to @jasmussen for the solution. It still needs to be tested in IE11. I tested with the latest version of Chrome, FF and Safari.

**Comment by harryfear** [comment]
Hi, folks,

The video settings only appear if there is a Media Library video at play, but not for YouTube videos:
https://ibb.co/zZBdQR7

Whereas pre-Gutenburg, the Poster Image setting was present even on YT videos:
https://www.wpoptimus.com/775/add-change-poster-image-videos-wordpress/

So the 'classic to Gutenburg' issue isn't fixed yet, as I can see.

**Comment by Soean** [comment]
@harryfear This was a Plugin (https://wordpress.org/plugins/youtube-chromeless/), it was not possible in a standard WordPress installation.

**Comment by harryfear** [comment]
I'm not 100% sure if that's right? A Chromeless YouTube player is possible now, and has been as of v. 3.9?

**Comment by strarsis** [comment]
For a `core/cover` block I cannot find the Video settings panel at all and no option for setting a poster/thumbnail.
The video currently doesn't load with a preview, which causes a kind of visual content flash.

**Pull request merged**